operator/v1/types_baremetal.go

```diff
@@ -0,0 +1,110 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3Provisioning contains configuration used by the Provisioning
+// service (Ironic) to provision baremetal hosts.
+//
+// Metal3Provisioning is created by the Openshift installer using admin
+// or user provided information about the provisioning network and the NIC
+// on the server that can be used to PXE boot it.
+//
+// This CR is a singleton, created by the installer and currently only
```

Comment:
@abhinavdahiya sounds like you'll have a change here.

@sadasu tell which name it is.  Should be "cluster" I'd think

Reply:
Referring to https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/baremetal-provisioning-config.md, we had decided upon main/default being the name.

Pasting (and re-formatting) from the above link:
"Important details of the CR:

Resource name - metal3provisioning.baremetal.operator.openshift.io
Instance name - main/default
Namespace - openshift-machine-api
Version - apiextensions.k8s.io/v1"